package/MDAnalysis/transformations/translate.py

```diff
@@ -21,50 +21,120 @@
-Translations --- :mod:`MDAnalysis.transformations.translate`
-============================================================
+Trajectory translation --- :mod:`MDAnalysis.transformations.translate`
+======================================================================
 
 Translate the coordinates of a given trajectory by a given vector.
-
-.. autofunction:: translate   
-    
+The vector can either be user defined, using the function :func:`translate`
+or defined by centering an AtomGroup in the unit cell using the function
+:func:`center_in_box`
     
 """
-from __future__ import absolute_import
+from __future__ import absolute_import, division
 
 import numpy as np
+from functools import partial
+
+from ..lib.mdamath import triclinic_vectors
+from ..core.groups import AtomGroup
 
 def translate(vector):
     """
     Translates the coordinates of a given :class:`~MDAnalysis.coordinates.base.Timestep`
     instance by a given vector.
     
+    Example
+    -------
+        ts = MDAnalysis.transformations.translate([1,2,3])(ts)    
+    
+    Parameters
+    ----------
+    vector: array-like
+        coordinates of the vector to which the coordinates will be translated
+        
+    Returns
+    -------
+    :class:`~MDAnalysis.coordinates.base.Timestep` object
+    
+    """
+    if len(vector)>2:
+        vector = np.float32(vector)
+    else:
+        raise ValueError("{} vector is too short".format(vector))
+    def wrapped(ts):
+        ts.positions += vector
+        
+        return ts
+    
+    return wrapped
+
+def center_in_box(ag, center='geometry', point=None, pbc=False):
+    """
+    Translates the coordinates of a given :class:`~MDAnalysis.coordinates.base.Timestep`
+    instance so that the center of geometry/mass of the given :class:`~MDAnalysis.core.groups.AtomGroup`
+    is centered on the unit cell. The unit cell dimensions are taken from the input Timestep object.
+    If a point is given, the center of the atomgroup will be translated to this point instead. 
+    
     Example
     -------
     
     .. code-block:: python
     
-        ts = MDAnalysis.transformations.translate([1,2,3])(ts)    
+        ag = u.residues[1].atoms
+        ts = MDAnalysis.transformations.center(ag,center='mass')(ts)
     
     Parameters
     ----------
-    vector: list
-        coordinates of the vector to which the coordinates will be translated
-    ts: :class:`~MDAnalysis.coordinates.base.Timestep`
-        frame that will be transformed
-     
+    ag: AtomGroup
+        atom group to be centered on the unit cell.
+    center: str, optional
+        used to choose the method of centering on the given atom group. Can be 'geometry'
+        or 'mass'
+    point: array-like, optional
+        overrides the unit cell center - the coordinates of the Timestep are translated so
+        that the center of mass/geometry of the given AtomGroup is aligned to this position
+        instead. Defined as an array of size 3.
+    pbc: bool, optional
```

Comment:
So you've been consistent, but imho [pbc is ambiguous](https://github.com/MDAnalysis/mdanalysis/issues/1185#issuecomment-360478631).  Could we call this `wrap` instead?